src/chttpd/test/exunit/pagination_test.exs

```diff
@@ -1163,7 +1221,153 @@ defmodule Couch.Test.Pagination do
+                           skip: 2
+                         }
+                       ]
+                     }
+        setup [:with_session, :random_db, :with_view, :with_same_key_docs]
+
+        test "handle same key", ctx do
+          '''
+          make sure the results are first sorted by key and then by id
+          '''
+          [descending_query, limit_query] =
+            paginate_queries(
+              ctx,
+              url:
+                "/#{ctx.db_name}/_design/#{ctx.ddoc_id}/_view/#{ctx.view_name}/queries",
+              query: %{page_size: ctx.page_size, descending: ctx.descending},
+              body: :jiffy.encode(ctx.queries)
+            )
+          IO.puts("descending_query 1: #{inspect descending_query}")
+          aggregate = fn pages ->
+            Enum.reduce(pages, {[], %{}}, fn page, acc ->
+              Enum.reduce(page, acc, fn row, {keys, in_acc} ->
+                id = Map.get(row, "id")
+                key = Map.get(row, "key")
+                {keys ++ [key], Map.update(in_acc, key, [id], &(&1 ++ [id]))}
+              end)
+            end)
+          end
+          {keys, aggregated} = aggregate.(descending_query)
+
+          # keys are sorted in reverse order
+          assert :descending == ordering?(keys),
+            ~s(expecing keys in descending order, got: #{inspect keys})
+
+
+          Enum.each(Map.values(aggregated), fn ids ->
+            # Keys are sorted in reverse order by id
+            assert :descending == ordering?(ids),
+              ~s(expecing ids in descending order, got: #{inspect ids})
+          end)
+
+          {keys, aggregated} = aggregate.(limit_query)
+          {expected_key_order, expected_ids_order} = if ctx.descending do
+            {:descending, :descending}
+          else
+            {:ascending, :ascending}
+          end
+
+          # keys are sorted
+          assert expected_key_order == ordering?(keys) or :equal == ordering?(keys),
+            ~s(expecing keys in #{expected_key_order} order, got: #{inspect keys})
+
+          Enum.each(Map.values(aggregated), fn ids ->
+            # Keys are sorted by id
+            assert expected_ids_order == ordering?(ids) or :equal == ordering?(ids),
+              ~s(expecing ids in #{expected_ids_order} order, got: #{inspect ids})
+          end)
+        end
+      end
+    end
+  end
+
+  defp ordering?(maps, key) do
+    ordering?(field(maps, key))
+  end
+
+  defp ordering?(elements) do
+    ascending = Enum.sort(elements)
+    descending = Enum.reverse(Enum.sort(elements))
+    '''
+    case elements do
+      ^ascending ->
+        :ascending
+      ^descending ->
+        :descending
+      _ ->
+        :unordered
+    end
+    '''
+    case {ascending, descending} do
+      {^elements, ^elements} -> :equal
+      {^elements, _} -> :ascending
+      {_, ^descending} -> :descending
+      _ -> :unordered
+    end
+  end
+
+  defp field(maps, key) do
+    Enum.map(maps, &(Map.get(&1, key)))
```

Comment:
Would `&Map.get(&1, key)` work here?

Reply:
Can you elaborate a little bit?

Reply:
The parentheses could be optional and they  are unambiguous in this case.


```elixir
> maps = [%{1=>2, 3=>4}, %{1=>6}, %{1=>7}]
[%{1 => 2, 3 => 4}, %{1 => 6}, %{1 => 7}]

> Enum.map(maps, &Map.get(&1, 1))
[2, 6, 7]
```

It's a minor nit, so whatever looks better to you